SpringProject/src/test/java/com/gradproject2019/UserServiceTest.java

```diff
@@ -15,4 +23,39 @@
 
     @Mock
     private UserRepository userRepository;
-}
+
+    private final User qwerty = new User( 3L, "qwerty", "qwerty", "qwerty", "qwerty@qwerty.com", "Qwerty!1", "qwerty");
+
+    @Test(expected = CredentialsExistException.class)
+    public void shouldThrowErrorWhenUsernameExists() {
+        //given user adds username
+        given(userRepository.findByUsername("qwerty")).willReturn(Optional.of(qwerty));
+        UserRequestDto copycat = createUserRequestDto("notqwerty","notqwerty", "notqwerty@qwerty.com", "notqwerty", "NotQwerty!1",  "qwerty");
+
+        //when username already exists
```

Comment:
Your tests say what you've written in the comments in this test file.

Reply:
On a more general note, if you code is clear enough to read (as it should be!) then you don't need any comments at all!